src/openms/source/CHEMISTRY/SvmTheoreticalSpectrumGenerator.cpp

```diff
@@ -961,24 +972,22 @@ namespace OpenMS
         Size j = 0;
         for (IsotopeDistribution::ConstIterator it = dist.begin(); it != dist.end(); ++it, ++j)
         {
-          p_.setMZ(mz_pos + (double)j * Constants::NEUTRON_MASS_U / charge);
-          p_.setIntensity(intensity * it->second);
-          if (add_metainfo && j == 0)
+          spectrum.push_back(Peak1D(mz_pos + (double)j * Constants::C13C12_MASSDIFF_U / charge, intensity * it->second));
+          if (add_metainfo)
           {
-            p_.setMetaValue("IonName", ion_name);
+            spectrum.getStringDataArrays()[0].push_back(ion_name);
+            spectrum.getIntegerDataArrays()[0].push_back(charge);
```

Comment:
you could change this to 
```
+            spectrum.getStringDataArrays()[ion_name_array].push_back(ion_name);
+            spectrum.getIntegerDataArrays()[ion_name_array].push_back(charge);
```

(see above) instead of assuming that its always the first array